### PR TITLE
fix(side_shift): side shift unable to properly reset

### PR DESCRIPTION
**Pull request by zulfaqar-azmi-t4**
Add comparison between lateral offset and shift point.

the last shift point was compared.

Signed-off-by: Muhammad Zulfaqar Azmi <zulfaqar.azmi@tier4.jp>

## Related Issue(required)

#441 

## Description(required)

Shifting happen (as the image below) even despite reset button was pressed.

The following is the current implementation of `updateStatus` function

This PR adds shift point with lateral offset comparison

And prevents continuous lateral offset change request by modifying
following `onLateralOffset` function

to 

## Review Procedure(required)

Require updated [configuration file](https://github.com/tier4/autoware_launch/pull/226)
1. Run FOA Console and Planning Simulator.
2. Select start position and goal.
3. Engage ego vehicle.
4. While ego vehicle is moving, perform side shift to the left and right again and again.
5. Then press reset to return shift point to the middle of the lane.
6. Let ego vehicle stay for some time to see whether there is any shifting occured.

## Related PR(optional)

[launch file](https://github.com/tier4/autoware_launch/pull/226)

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [ ] Read [commit-guidelines][commit-guidelines]
- [ ] Assign PR to reviewer

If you are adding new package following items are required:

- [ ] Documentation with description of the package is available
- [ ] A sample launch file and parameter file are available if the package contains executable nodes

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[commit-guidelines]: https://www.conventionalcommits.org/en/v1.0.0/
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: planning/behavior_path_planner/include/behavior_path_planner/scene_module/side_shift/side_shift_module.hpp
Comment (yukkysaito) [inline planning/behavior_path_planner/include/behavior_path_planner/scene_module/side_shift/side_shift_module.hpp:59]: What is the unit? msec?

min_request_time_msec might be better.

File: planning/behavior_path_planner/include/behavior_path_planner/scene_module/side_shift/side_shift_module.hpp
Comment (yukkysaito) [inline planning/behavior_path_planner/include/behavior_path_planner/scene_module/side_shift/side_shift_module.hpp:63]: If the name is requested, I think it's bool.
interval_from_last_time_msec is better, maybe

File: planning/behavior_path_planner/src/scene_module/side_shift/side_shift_module.cpp
Comment (yukkysaito) [inline planning/behavior_path_planner/src/scene_module/side_shift/side_shift_module.cpp:77]:
```suggestion
  request_timer = SideShiftRequestTimer();
```

File: planning/behavior_path_planner/include/behavior_path_planner/scene_module/side_shift/side_shift_module.hpp
Comment (zulfaqar-azmi-t4) [reply planning/behavior_path_planner/include/behavior_path_planner/scene_module/side_shift/side_shift_module.hpp:63]: Thank you. yes, `interval_from_last_time_msec` is much better.

File: planning/behavior_path_planner/include/behavior_path_planner/scene_module/side_shift/side_shift_module.hpp
Comment (yukkysaito) [inline planning/behavior_path_planner/include/behavior_path_planner/scene_module/side_shift/side_shift_module.hpp:53]: The name(SideShiftRequestTimer) is supposed to send requests periodically, is that correct?

File: planning/behavior_path_planner/include/behavior_path_planner/scene_module/side_shift/side_shift_module.hpp
Comment (yukkysaito) [inline planning/behavior_path_planner/include/behavior_path_planner/scene_module/side_shift/side_shift_module.hpp:136]:
```suggestion
  SideShiftRequestTimer request_timer_;
```
It is private var :+1:

File: planning/behavior_path_planner/include/behavior_path_planner/scene_module/side_shift/side_shift_module.hpp
Comment (zulfaqar-azmi-t4) [reply planning/behavior_path_planner/include/behavior_path_planner/scene_module/side_shift/side_shift_module.hpp:59]: the time value should be in second, i.e: 0.5 second or 1.0 second, so that it sync with `time_to_start_shifting` from `side_shift.param.yaml`

File: planning/behavior_path_planner/include/behavior_path_planner/scene_module/side_shift/side_shift_module.hpp
Comment (zulfaqar-azmi-t4) [reply planning/behavior_path_planner/include/behavior_path_planner/scene_module/side_shift/side_shift_module.hpp:53]: The `SideShiftRequestTimer` doesn't send any request. It is just used track `lateral_offset` requests made by the scene module node.

**Comment by zulfaqar-azmi-t4** [comment]
Seems like it is best for the request time to be configurable, so I've added a new parameters for this.

File: planning/behavior_path_planner/src/scene_module/side_shift/side_shift_module.cpp
Comment (1222-takeshi) [inline planning/behavior_path_planner/src/scene_module/side_shift/side_shift_module.cpp:205]: Is this change separate from this bug?

Comment (zulfaqar-azmi-t4) [reply planning/behavior_path_planner/src/scene_module/side_shift/side_shift_module.cpp:205]: The change does not directly related to the bug.|
However, it very slightly increase the performance and it should allow the shift point above threshold to be remove more "smoothly", which may result in more stable path.

Comment (zulfaqar-azmi-t4) [reply planning/behavior_path_planner/src/scene_module/side_shift/side_shift_module.cpp:205]: However, removing this changes will not affect how the bug is fixed, therefore, I can always revert this part

**Review by 1222-takeshi — APPROVED** [review]
LGTM

**Comment by 1222-takeshi** [comment]
@zulfaqar-azmi-t4 
saito-san's part of the review, please discuss with @yukkysaito to resolve the issue . :bow:

**Comment by yukkysaito** [comment]
Please wait until @TakaHoribe and @kenji-miyake approve.

**Comment by kenji-miyake** [comment]
I'll wait for comments from Taka for now.

**Comment by TakaHoribe** [comment]
Sorry for the late response. I've heard this is not an urgent issue, but I hope I could take it in a few days.

File: planning/behavior_path_planner/src/scene_module/side_shift/side_shift_module.cpp
Comment (TakaHoribe) [inline planning/behavior_path_planner/src/scene_module/side_shift/side_shift_module.cpp:336]: @zulfaqar-azmi-t4 thank you so much for the fix

File: planning/behavior_path_planner/src/scene_module/side_shift/side_shift_module.cpp
Comment (TakaHoribe) [inline planning/behavior_path_planner/src/scene_module/side_shift/side_shift_module.cpp:371]: Great improvement.

**Comment by TakaHoribe** [comment]
@zulfaqar-azmi-t4 Sorry for the late response. I checked this modification and it seems fine to me. One thing, please use RCLCPP_TIME instead of the Chrono to make a consistency when we use the ROS clock.

File: planning/behavior_path_planner/include/behavior_path_planner/scene_module/side_shift/side_shift_module.hpp
Comment (zulfaqar-azmi-t4) [inline planning/behavior_path_planner/include/behavior_path_planner/scene_module/side_shift/side_shift_module.hpp:118]: @TakaHoribe 
`clock_->now()` is used to replace `chrono`

**Comment by tkimura4** [comment]
@TakaHoribe 
Is it ready for merging?

**Review by TakaHoribe — APPROVED** [review]
Then, LGTM!

**Pull request merged**
